.github/workflows/README.md

```diff
@@ -5,6 +5,152 @@ CI system guide
+    end
+    subgraph cpp_tests
+        direction LR
+        cpp_tests_windows_compile --> cpp_tests_windows
+        cpp_tests_linux_compile --> cpp_tests_linux
+    end
+    subgraph NOT YET IMPLEMETED
+        subgraph real_storage_tests
+            direction LR
+            pre_real_storages_test --> real_storages_test
+            real_storages_test --> post_real_storages_test
+        end
+    end
+    leader --> follower
+    leader --> cpp_tests
+    leader --> real_storage{Should test against real storages?} --> real_storage_tests --> can_merge
+    follower --> docs
+    docs --> can_merge
+    cpp_tests --> can_merge
+    can_merge --> pub_check{publish_env} --> publish
+```
+
+This diagram overviews the structure of the main CI system. \
+The system is outlined in the [build.yml](build.yml) file. \
+The concrete steps are implemented in [build_steps.yml](build_steps.yml) file. \
+For more information, see the [Description of the YML files](#description-of-the-yml-files) section.
+
+## Common config job
+
+This job is used to configure the CI run. \
+It sets up what subsequent jobs, their structure where they will be execute, etc. \
+This is where environemt variables are set and also some of the variables that are used by the later jobs. \
+You can see them in the outputs part and they are used by the other jobs by calling *needs.common_config.outputs...*.
+
+## CI Build Wheel Docker Image
+
+TODO
+
+## Leader jobs
+
+The leader jobs are designed to do the C++ core compilation using one Python version to seed the compilation caches. \
+This way, if there is a general problem with the build, it can fail quicker, without having to way for the different versions.
+There are leader jobs for both Linux and Windows.
+
+## C++ Tests jobs
+
+These jobs compile the C++ code and run the C++ tests. \
+They are designed to run concurently with the Follower jobs, which test against the different versions of Python.
+
+## Follower jobs
+
+After we the leader jobs have passed succesfully, we run Follower jobs that compile and test against the rest of the Python versions that are supported.
+
+## Docs job
+
+The Docs job compiles and publishes the latest docs. \
+It needs a valid ArcticDB wheel to run, so it depedns on the Linux jobs. \
+Also currently, the automatic builds trigger only from cahnges in the code. \
+**So if you make changes that are only in the docs, you will need to start a manual build and supply a valid ArcticDB wheel.**
+
+## can_merge check
+
+This is a simple check that is used by GitHub to determine, if a PR can be merged. \
+It is also used as a gate to check, if we can publish a new release.
+
+## Publish
+
+In this step, we prepare the release notes and publish to PyPi. \
+This step cannot run on automatic builds, because it requires the *pypi_publish* argument to be enable, which can be done only with a manual build.
+
+# Build Inputs
+
+When a manual build is trigger, the user can specify the following input arguments:
```

Comment:
```suggestion
When a manual build is triggered, the user can specify the following input arguments:
```

I strongly recommend documenting these in the yaml itself and linking to the lines.